encrypted-payloads/crypt-converter.go

```diff
@@ -0,0 +1,150 @@
+package cryptconverter
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"fmt"
+	"io"
+
+	commonpb "go.temporal.io/api/common/v1"
+
+	"go.temporal.io/sdk/converter"
+)
+
+// CryptDataConverter implements DataConverter using AES Crypt.
+type CryptDataConverter struct {
+	dataConverter converter.DataConverter
+}
+
+// getKey fetches the crypt key from secure storage
+func (dc *CryptDataConverter) getKey() (key []byte) {
+	// Key can be fetched from KMS or other secure storage.
+	return []byte("test-key-test-key-test-key-test!")
+}
+
+// NewCryptDataConverter created new instance of CryptDataConverter wrapping a DataConverter
+func NewCryptDataConverter(dataConverter converter.DataConverter) *CryptDataConverter {
+	return &CryptDataConverter{
+		dataConverter: dataConverter,
+	}
+}
+
+func encrypt(plaintext []byte, key []byte) ([]byte, error) {
+	c, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCM(c)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, err
+	}
+
+	return gcm.Seal(nonce, nonce, plaintext, nil), nil
+}
+
+func decrypt(ciphertext []byte, key []byte) ([]byte, error) {
```

Comment:
```suggestion
func decrypt(cipherData []byte, key []byte) ([]byte, error) {
```

Reply:
Just FYI I would find this weird. I have never seen anyone call it `cypherData`, even if maybe that makes more sense. It's always `cypherText` IME -- or `encrypted` or `encryptedData`, which is probably what I'd go with here.